### DISCUSSION TITLE
Code Smartness on [ObjectApiName].SObjectType

### DISCUSSION THREAD

**Discussion by ONGChain** [forcedotcom/salesforcedx-vscode#5476] [Ideas]
**Is your feature request related to a problem? Please describe.**
I find it very frustrating that code smartness and VS code quick actions do not work with the shorthand tokens for Schema.SObjectType and Schema.SObjectField. 

For example, if a method call is passing the field token Account.SObjectType as an argument. I am then unable to CTRL + Click the method name to go to the method declaration as the code smartness does not recognize Account.SObjectType as a Schema.SObjectType despite the code executing just fine. This would be the same for if it was a Schema.SObjectField token.

I'm currently working with another dev's code and there are a lot of SObjectTypes and SObjectFields being passed around in their implementation using this shorthand syntax. Being unable to quickly go to the declarations is slowing down my workflow. Also, would prefer to use the token style as opposed to the method style of Account.getSObjectType() or Account.Name.getDescribe() as I find it much cleaner looking since logically a SObjectType or Field is just a property of a SObject.

**Describe the solution you'd like**
I would like for the SObjectType and SObjectField tokens to be recognized as their respective types so that the benefits of using the VS code IDE can be utilized.

**Describe alternatives you've considered**
I've tried adding the tokens as properties into the classes generated by the refresh SObjects command, but that was unsuccessful. I could do a search and replace via Regex for all instances of the tokens to convert them to their method counterparts, but I feel updating the extension is the only correct answer to this problem.

**Additional context**

![image](https://user-images.githubusercontent.com/115097388/221716118-3f474dc3-88b9-4a65-9919-dcf3d89e1618.png)
![image](https://user-images.githubusercontent.com/115097388/221716227-6d229dc7-cc31-4fea-8a2e-66c5501ca9a6.png)

**Comment by git2gus** [comment]
This issue has been linked to a new work item: W-13465519
